File: src/main/java/org/cloudfoundry/credhub/entity/JsonCredentialVersionData.java
package org.cloudfoundry.credhub.entity;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Entity
@DiscriminatorValue(JsonCredentialVersionData.CREDENTIAL_TYPE)
public class JsonCredentialVersionData extends CredentialVersionData<JsonCredentialVersionData> {

  public static final String CREDENTIAL_TYPE = "json";

  public JsonCredentialVersionData() {
  }

  public JsonCredentialVersionData(String name) {
    super(name);
  }

  @Override
  public String getCredentialType() {
    return CREDENTIAL_TYPE;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/entity/JsonCredentialVersionData.kt
package org.cloudfoundry.credhub.entity

import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity

@Entity
@DiscriminatorValue(JsonCredentialVersionData.CREDENTIAL_TYPE)
class JsonCredentialVersionData : CredentialVersionData<JsonCredentialVersionData> {
    override val credentialType: String
        get() = CREDENTIAL_TYPE

    constructor() : super() {}

    constructor(name: String) : super(name) {}

    companion object {
        const val CREDENTIAL_TYPE = "json"
    }
}
